eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml

```diff
@@ -1,18 +1,18 @@
 steps:
+  - ${{if eq(variables['DefaultBranch'], '')}}:
+    - template: /eng/common/pipelines/templates/steps/set-default-branch.yml
+      parameters:
+        DefaultBranchVariableName: DefaultBranch
   - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
     - pwsh: |
         Set-PsDebug -Trace 1
         # Find the default branch of the repo
-        $defaultBranch = (git remote show origin | Out-String) -replace "(?ms).*HEAD branch: (\w+).*", '$1'
+        Write-Host "Default Branch: $(DefaultBranch)"
         $targetBranch = "$(System.PullRequest.TargetBranch)" -replace "refs/heads/"
 
-        Write-Host "Default Branch: ${defaultBranch}"
-
-        if ((!"$(System.PullRequest.SourceBranch)".StartsWith("sync-eng/common")) -and $targetBranch -eq $defaultBranch)
+        if ((!"$(System.PullRequest.SourceBranch)".StartsWith("sync-eng/common")) -and $targetBranch -eq "$(DefaultBranch)")
         {
-
-          $filesInCommonDir = git diff "origin/${targetBranch}" HEAD --name-only -- 'eng/common/*'
-
+          $filesInCommonDir  = & "eng/common/scripts/git-diff-changes.ps1" -DiffPath 'eng/common/*'
```

Comment:
Lets pass targetBranch here just to ensure things remain consistent with the local logic.